eZ/Publish/Core/MVC/Symfony/Cache/Http/Proxy/TagAwareStore.php

```diff
@@ -0,0 +1,241 @@
+<?php
+
+/**
+ * File containing the TagAwareStore class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ *
+ * @version //autogentag//
+ */
+namespace eZ\Publish\Core\MVC\Symfony\Cache\Http\Proxy;
+
+use eZ\Publish\Core\MVC\Symfony\Cache\Http\ContentPurger;
+use Symfony\Component\HttpKernel\HttpCache\Store;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Filesystem\Exception\IOException;
+use Symfony\Component\Finder\Finder;
+
+/**
+ * TagAwareStore implements all the logic for storing cache metadata regarding tags (locations, content type, ..).
+ */
+class TagAwareStore extends Store implements ContentPurger
+{
+    const TAG_CACHE_DIR = 'ez';
+
+    /**
+     * @var \Symfony\Component\Filesystem\Filesystem
+     */
+    private $fs;
+
+    /**
+     * Injects a Filesystem instance
+     * For unit tests only.
+     *
+     * @internal
+     *
+     * @param \Symfony\Component\Filesystem\Filesystem $fs
+     */
+    public function setFilesystem(Filesystem $fs)
+    {
+        $this->fs = $fs;
+    }
+
+    /**
+     * @return \Symfony\Component\Filesystem\Filesystem
+     */
+    public function getFilesystem()
+    {
+        if (!isset($this->fs)) {
+            $this->fs = new Filesystem();
+        }
+
+        return $this->fs;
```

Comment:
As in why do we need `getFilesystem()`? I'll have to check, as many of your other questions it's inherited from the previous class.